### ISSUE TITLE
Multiple warnings and a notice on saving config

### ISSUE THREAD

**Issue by yorkshire-pudding** [backdrop-contrib/google_tag#2]
**Steps to reproduce**

1. Ensure warnings and notices set to display
2. Enable Google Tag manager module
3. Go to configure
4. Enter the tag
5. Save configuration

**Expected result**
Saves with no errors, warnings or notices

**Actual result**
Config is saved but warnings and a notice appears:

<details>

Warning: Parameter 3 to google_tag_validate_text() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).   

Warning: Parameter 3 to google_tag_validate_text() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).  

Warning: Parameter 3 to google_tag_validate_data_layer() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).  

Warning: Parameter 3 to google_tag_validate_listed_classes() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).  

Warning: Parameter 3 to google_tag_validate_text() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).  

Warning: Parameter 3 to google_tag_validate_text() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).  

Warning: Parameter 3 to google_tag_validate_environment_id() expected to be a reference, value given in _form_validate() (line 1493 of /app/docroot/core/includes/form.inc).  

Notice: Undefined index: google_tag_data_layer_classes in google_tag_settings_form_submit() (line 328 of /app/docroot/modules/contrib/google_tag/google_tag.admin.inc).  
</details>

**Comment by laryn** [comment]
@yorkshire-pudding I've made a PR here -- interested to test? I'm not sure why those validation functions are completely commented out, but I think they are still getting called and their parameters were not what was expected in a validation callback, I think.

**Comment by yorkshire-pudding** [comment]
@laryn - I don't think any of my projects use this anymore.  The project I was working on when I raised this is using the Google Analytics module and all the others I'm using are on Matomo.

However, I've downloaded the dev version of the project into a test site, enabled all messages and recreated the issue with a dummy tag.  I then downloaded and applied the patch and can confirm it fully resolves the issue.

Thank you
